README.md

```diff
@@ -112,52 +136,111 @@ func main() {
+for OAuth client credentials in environment variables and then falls back to
+Azure's [Managed Service Identity]() when available, e.g. when on an Azure
+VM. The following snippet from [the previous section](#use) demonstrates
+this helper.
+
+```go
+import github.com/Azure/go-autorest/autorest/azure/auth
+
+// create a VirtualNetworks client
+vnetClient := network.NewVirtualNetworksClient("<subscriptionID>")
+
+// create an authorizer from env vars or Azure Managed Service Idenity
+authorizer, err := auth.NewAuthorizerFromEnvironment()
+if err == nil {
+    vnetClient.Authorizer = authorizer
+}
+
+// call the VirtualNetworks CreateOrUpdate API
+vnetClient.CreateOrUpdate(context.Background(),
+// ...
+```
+
+The following environment variables help determine authentication configuration:
+
+- `AZURE_ENVIRONMENT`: Specifies the Azure Environment to use. If not set, it
+  defaults to `AzurePublicCloud`. Not applicable to authentication with Managed
+  Service Identity (MSI).
+- `AZURE_AD_RESOURCE`: Specifies the AAD resource ID to use. If not set, it
+  defaults to `ResourceManagerEndpoint` for operations with Azure Resource
+  Manager.
+
+## Authentication Details
+
+The previous is the first and most recommended of several authentication
+options offered by the SDK because it allows seamless use of both service
+principals and [Azure Managed Service Identity][]. Other options are listed
+below.
+
+> Note: If you need to create a new service principal, run `az ad sp
+> create-for-rbac -n "<app_name>"` in the
+> [azure-cli](https://github.com/Azure/azure-cli). See [these
+> docs](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli?view=azure-cli-latest)
+> for more info. Copy the new principal's ID, secret, and tenant ID for use in
+> your app, or consider the `--sdk-auth` parameter for serialized output.
+
+[Azure Managed Service Identity]: https://docs.microsoft.com/en-us/azure/active-directory/msi-overview
+
+* The `auth.NewAuthorizerFromEnvironment()` described above creates an authorizer
+from the first available of the following configuration:
+
+    1. **Client Credentials**: Azure AD Application ID and Secret.
+
+        - `AZURE_TENANT_ID`: Specifies the Tenant to which to authenticate.
+        - `AZURE_CLIENT_ID`: Specifies the app client ID to use.
+        - `AZURE_CLIENT_SECRET`: Specifies the app secret to use.
+
+    2. **Client Certificate**: Azure AD Application ID and X.509 Certificate.
+
+        - `AZURE_TENANT_ID`: Specifies the Tenant to which to authenticate.
+        - `AZURE_CLIENT_ID`: Specifies the app client ID to use.
+        - `AZURE_CERTIFICATE_PATH`: Specifies the certificate Path to use.
+        - `AZURE_CERTIFICATE_PASSWORD`: Specifies the certificate password to use.
+
+    3. **Resource Owner**: Azure AD User and Password. This grant type is *not
+       recommended*, use client credentials (service principals) instead.
```

Comment:
maybe add that if this is desired for interactive user access they can use Device Flow for 2FA access